### PR TITLE
Clean api http

### PR DESCRIPTION
**Pull request by ghost**
- new routes for zenodo use cases
- better use of http verbs
- minor bug fixes
- updated tests

### CODE REVIEW

File: shipper.py [hunk 1]
```diff
@@ -86,8 +86,84 @@ def shipment_get_all():
...
+        data = db['shipments'].find_one({'id': shipmentid})
+        if data is not None:
+            if 'status' in data:
+                response.status = 200
+                return {'id': shipmentid, 'status': str(data['status'])}
+            else:
+                response.status = 400
+                return {'error': 'shipment data incomplete'}
+    except:
+        raise
```
Comment (nuest) [inline shipper.py]: Is the execption raised here actually encapsulated in a JSON document? Just wondering.

File: shipper.py [hunk 2]
```diff
@@ -86,8 +86,84 @@ def shipment_get_all():
...
+def shipment_get_status(shipmentid):
+    try:
+        data = db['shipments'].find_one({'id': shipmentid})
+        if data is not None:
+            if 'status' in data:
+                response.status = 200
+                return {'id': shipmentid, 'status': str(data['status'])}
+            else:
+                response.status = 400
+                return {'error': 'shipment data incomplete'}
```
Comment (nuest) [inline shipper.py]: "data incomplete" might be misleading. isn't it "shipment not found" ?

File: shipper.py [hunk 3]
```diff
@@ -86,8 +86,84 @@ def shipment_get_all():
...
+            if 'files' in r.json():
+                response.status = 200
+                return json_dumps({'files': r.json()['files']})
+            else:
+                response.status = 400
+                return {'error': 'no files object in repository response'}
+        elif db_find_recipient_from_shipment(shipmentid) == 'eudat':
+            pass
+        else:
+            status_note('unknown recipient')
```
Comment (nuest) [inline shipper.py]: is there no response to the user in this case?

File: shipper.py [hunk 4]
```diff
@@ -86,8 +86,84 @@ def shipment_get_all():
...
+            headers = {"Content-Type": "application/json"}
+            r = requests.get(''.join((env_repository_zenodo_host, '/deposit/depositions/', current_depot, '?access_token=', env_repository_zenodo_token)), headers=headers)
+            if 'files' in r.json():
+                response.status = 200
+                return json_dumps({'files': r.json()['files']})
+            else:
+                response.status = 400
+                return {'error': 'no files object in repository response'}
+        elif db_find_recipient_from_shipment(shipmentid) == 'eudat':
+            pass
```
Comment (nuest) [inline shipper.py]: is there no response to the user in this case?

File: shipper.py [hunk 5]
```diff
@@ -86,8 +86,84 @@ def shipment_get_all():
...
+                db.shipments.update_one({'id': shipmentid}, {'$set': {'status': 'published'}}, upsert=True)
+                if 'doi_url' in r.json():
+                    db.shipments.update_one({'id': shipmentid}, {'$set': {'doi_url': r.json()['doi_url']}}, upsert=True)
+                response.status = r.status_code
+                response.content_type = 'application/json'
+                return {'id': shipmentid, 'status': 'published'}
+            else:
+                response.status = r.status_code
+        elif db_find_recipient_from_shipment(shipmentid) == 'eudat':
+            pass
```
Comment (nuest) [inline shipper.py]: should return error if not implemented yet.

File: shipper.py [hunk 6]
```diff
@@ -86,8 +86,84 @@ def shipment_get_all():
...
+
+@app.route('/api/v1/shipment/<shipmentid>/files/<fileid>', method='DELETE')
+def shipment_del_file_id(shipmentid, fileid):
+    # delete specific of a depot of a shipment
+    try:
+        current_depot = db_find_depotid_from_shipment(shipmentid)
+        if db_find_recipient_from_shipment(shipmentid) == 'zenodo':
+            zenodo_del_from_depot(env_repository_zenodo_host, current_depot, fileid, env_repository_zenodo_token)
+        elif db_find_recipient_from_shipment(shipmentid) == 'eudat':
+            pass
```
Comment (nuest) [inline shipper.py]: same here, return error to user.

File: shipper.py [hunk 7]
```diff
@@ -500,7 +543,7 @@ def zen_del_from_depot(base, deposition_id, file_id, token):
         #status_note(''.join(('! error: ', exc.args[0])))
 
 
-def zen_del_depot(base, deposition_id, token):
+def zenodo_del_depot(base, deposition_id, token):
```
Comment (nuest) [inline shipper.py:556]: I like you using `zenodo_` instead of `zen_`!

File: shipper.py [hunk 8]
```diff
@@ -540,14 +601,14 @@ def xstr(s):
...
 if __name__ == "__main__":
-    my_version = 7  # update me!
-    my_mod = ''
-    try:
-        my_mod = datetime.fromtimestamp(os.stat(__file__).st_mtime)
-    except OSError as exc:
-        status_note(''.join(('! error: ', exc.args[0], '\n', traceback.format_exc())))
-        sys.exit(1)
-    status_note(''.join(('v', str(my_version), ' - ', str(my_mod))))
+    # my_version = 7  # update me! ## obsolete with microbadger build hash
```
Comment (nuest) [inline shipper.py]: if obsolete then I recommend to remove, not comment out. it will still be in git history :-)

**Review by nuest — COMMENTED** [review]
you already got some useful tests, imho it would only be a very small step towards integration tests (https://bottlepy.org/docs/dev/recipes.html#functional-testing-bottle-applications)

I understand though that the actual test if something is _published_ on a shipment recipient should not happen in an integration test... maybe you can fake that, but that might be too much.

File: shipper.py (hunk 2 above)
Comment (ghost) [reply shipper.py]: right

File: shipper.py (hunk 1 above)
Comment (ghost) [reply shipper.py]: no, it produces a server side exception.
I use it in debugging. must be replaced

File: shipper.py (hunk 4 above)
Comment (ghost) [reply shipper.py]: TBD

File: shipper.py (hunk 3 above)
Comment (ghost) [reply shipper.py]: TBD

File: shipper.py (hunk 5 above)
Comment (ghost) [reply shipper.py]: TBD

File: shipper.py (hunk 7 above)
Comment (ghost) [reply shipper.py:556]: slowly relocating from draftyness

File: shipper.py (hunk 8 above)
Comment (ghost) [reply shipper.py]: true

**Pull request merged**